core/state_processor.go

```diff
@@ -367,4 +368,6 @@ func ProcessParentBlockHash(statedb *state.StateDB, prevNumber uint64, prevHash
 	var key common.Hash
 	binary.BigEndian.PutUint64(key[24:], prevNumber)
 	statedb.SetState(params.HistoryStorageAddress, key, prevHash)
+	index, suffix := utils.GetTreeKeyStorageSlotTreeIndexes(key[:])
+	statedb.Witness().TouchAddressOnWriteAndComputeGas(params.HistoryStorageAddress[:], *index, suffix)
```

Comment:
Makes sense. At some point I wonder if it would be better to tie access witness recording with `statedb` methods that touch the state.

Just an idea though so these kind of misses are avoided.

Reply:
yeah it's something I've wanted to do for a long time. I tried once in the past but it was buggy and I had other priorities. I would wait for the shadowfork + rebase to do that, but it would be a pretty good thing.